[#108] Improve LIGO version documentation (#120)
tqtezos/baseDAO: ligo/README.md

# BaseDAO LIGO

__Note: This is still current in development.__

> This is the LIGO implementation of BaseDAO.

Reviewer: ```suggestion
This is the LIGO implementation of BaseDAO, written in the cameLIGO dialect.
```

The main difference between the two are:
- `BaseDAO Lorentz`:
    - uses compile-time configuration to configure various options of baseDAO, which means that each contract that uses `BaseDAO Lorentz` will have its own `michelson` contract.

- `BaseDAO LIGO`:
    - uses runtime configuration
    - only one `michelson` contract is used, the user simply defines the configuration as part of originated storage

> ## Prerequisites
> 
> You will need [ligo executable](https://ligolang.org/docs/intro/installation) installed.
> 
> ## Build instructions

Reviewer: I suggest moving `ligo` to be a build dependency:
```suggestion
## Build instructions

You will need the [ligo executable](https://ligolang.org/docs/intro/installation) installed to generate the contract.
```

> To build the LIGO contract, simply run:

Reviewer: ```suggestion
You can use the [`Makefile`](./Makefile) to build the LIGO contract by simply running:
```

```sh
make all
```

> 

Reviewer: ```suggestion
which will save the compiled contract in `out/baseDAO.tz`.
```

## Generating `Storage`
TODO

## Originating the contract
To originate `baseDAO LIGO`, run:
```

> TODO

Author: @pasqu4le Do we have a sample storage for LIGO dao? Since to originate the contract, we need a storage.

Hope to get review on the rest of the change as well.

Reviewer: I'm not sure we'll end up providing one at all, hence why the storage section is a TODO for now.

I suggested below in a comment how I think we should handle this.

> ## Originating the contract
> To originate `baseDAO LIGO`, run:
> ```
> TODO
> ```

Reviewer: Here instead I suggest mentioning `tezos-client` as a possible choice only.

I don't think it's a good idea to write an actual storage in this README, let's only mention an example call.
```suggestion
## Using the contract

You will need [tezos-client](http://tezos.gitlab.io/introduction/howtoget.html) or similar software to originate and interact with the contract on the Tezos network.

For example, if you have compiled the contract and obtained a storage (`<storage>`) as described above you can originate it with:
`tezos-client originate contract myDAO transferring 1 from alice running out/baseDAO.tz --init <storage>`
```

## Testing

> `baseDAO LIGO` shares the tests with `baseDAO Lorentz` which are implemented in Haskell.
> 
> To run the test:

Reviewer: I think it's not necessary to repeat this here, let's just leave the line below pointing at the test's README for more info:
```suggestion
The tests for this contract are written in Haskell and you will need the `stack` tool to run them.

You can use the `Makefile` to do so as well:
```
